extensions-core/s3-extensions/src/main/java/io/druid/storage/s3/S3TimestampVersionedDataFinder.java

```diff
@@ -67,7 +67,7 @@ public URI call() throws Exception
               final S3Coords coords = new S3Coords(checkURI(uri));
               long mostRecent = Long.MIN_VALUE;
               URI latest = null;
-              S3Object[] objects = s3Client.listObjects(coords.bucket, coords.path, "/");
+              S3Object[] objects = s3Client.listObjects(coords.bucket, coords.path, null);
```

Comment:
The old behavior was correct according to the javadoc, and was consistent with the hdfs & local pullers. If this new behavior is better, we should probably update the javadoc and perhaps the other pullers. I can sorta see an argument why the S3 puller should be different (S3 doesn't really have directories) but even so, many (most?) tools for interacting with S3 still pretend it has directories anyway.